dataflow/src/main/java/org/checkerframework/dataflow/util/PurityUtils.java

```diff
@@ -98,36 +98,64 @@ public static boolean isSideEffectFree(AnnotationProvider provider, MethodTree m
    */
   public static boolean isSideEffectFree(
       AnnotationProvider provider, ExecutableElement methodElement) {
-    EnumSet<Pure.Kind> kinds = getPurityKinds(provider, methodElement);
+    EnumSet<Pure.Kind> kinds = getPurityAnnotations(provider, methodElement);
     return kinds.contains(Kind.SIDE_EFFECT_FREE);
   }
 
   /**
-   * Returns the types of purity of the method {@code methodTree}.
+   * Returns the purity annotations of the method {@code methodTree}.
    *
    * @param provider how to get annotations
    * @param methodTree a method to test
    * @return the types of purity of the method {@code methodTree}
+   * @deprecated use {@code getPurityAnnotations}
    */
+  @Deprecated // 2022-09-27
   public static EnumSet<Pure.Kind> getPurityKinds(
       AnnotationProvider provider, MethodTree methodTree) {
+    return getPurityAnnotations(provider, methodTree);
+  }
+
+  /**
+   * Returns the purity annotations on the method {@code methodTree}.
+   *
+   * @param provider how to get annotations
+   * @param methodTree a method to test
+   * @return the types of purity of the method {@code methodTree}
+   */
+  public static EnumSet<Pure.Kind> getPurityAnnotations(
+      AnnotationProvider provider, MethodTree methodTree) {
     ExecutableElement methodElement = TreeUtils.elementFromDeclaration(methodTree);
     if (methodElement == null) {
       throw new BugInCF("Could not find element for tree: " + methodTree);
     }
-    return getPurityKinds(provider, methodElement);
+    return getPurityAnnotations(provider, methodElement);
   }
 
   /**
-   * Returns the types of purity of the method {@code methodElement}.
+   * Returns the purity annotations on the method {@code methodElement}.
    *
    * @param provider how to get annotations
    * @param methodElement a method to test
    * @return the types of purity of the method {@code methodElement}
+   * @deprecated use {@code getPurityAnnotations}
    */
-  // TODO: should the return type be an EnumSet?
+  @Deprecated // 2022-09-27
   public static EnumSet<Pure.Kind> getPurityKinds(
       AnnotationProvider provider, ExecutableElement methodElement) {
+    return getPurityAnnotations(provider, methodElement);
+  }
+
+  /**
+   * Returns the types of purity of the method {@code methodElement}.
+   *
+   * @param provider how to get annotations
+   * @param methodElement a method to test
+   * @return the types of purity of the method {@code methodElement}
+   */
+  public static EnumSet<Pure.Kind> getPurityAnnotations(
```

Comment:
Given this method name, I would expect it to return a set (or collection) of `AnnotationMirror`s.  The previous name  reflected that set of `Pure.Kind` is returned.  So, I wouldn't rename this method.